api/explorer/api/ingest.py

```diff
@@ -98,72 +97,18 @@ def _edge_from_build_input_type(build_input_type: model.BuildInputType) -> graph
-    # input node into N nodes based on the number of the node's output paths.
-    pkgs = safe_parse_package(model_pkg)
-    # Create all dependency nodes and draw an edge between the input package nodes and
-    # them.
-    for pkg in pkgs:
-        for build_input in model_pkg.build_inputs:
-            edge = _edge_from_build_input_type(build_input.build_input_type)
-            try:
-                # Since build_inputs is defined using the core model, we need to parse
-                # it into the graph model similar to what we do above for model_pkg.
-                build_input_pkgs = safe_parse_package(build_input.package)
-            except IngestionError as e:
-                logger.exception(e)
-                build_input_pkgs = []
-            for bi_pkg in build_input_pkgs:
-                do_fn(pkg, bi_pkg, edge)
-    return [bi.package for bi in model_pkg.build_inputs]
-
-
-A = TypeVar("A")
-
-
-def flatten(lls: Iterable[Iterable[A]]) -> list[A]:
-    """Flattens nested iterables into a list"""
-    return [x for sublist in lls for x in sublist]
-
-
-def traverse(
-    nix_graph: model.NixGraph,
-    root_fn: Callable[[graph.Package], None],
-    do_fn: Callable[[graph.Package, graph.Package, graph.Edge], None],
-):
-    """Traverse a Nix graph
-
-    Performs a breadth-first traversal of a nix graph, executing the provided
-    functions on each level of the graph.
-
-    Args:
-        nix_graph (model.NixGraph): The NixGraph to traverse
-        root_fn (Callable[[graph.Package], None]): A function to call only on the root
-            level of the nix graph. This can be useful for things like initialization
-            steps.
-        do_fn (Callable[[graph.Package, graph.Package, graph.Edge], None]): A function
-            to call on every level of the graph, including the root level. This
-            function is expected to take three arguments which correspond to the
-            current level, the next level, and the edge describing the relationship
-            between the current level and the next level.
     """
-    for model_pkg in nix_graph.packages:
-        # Apply the user provided function to the first layer
-        pkgs = safe_parse_package(model_pkg)
-        for pkg in pkgs:
-            root_fn(pkg)
-        next_pkgs = _do_next_graph_level(model_pkg, do_fn)
-        # Iterate over each remaining layer until no more remain
-        while next_pkgs:
-            # TODO: Exception handling
-            next_pkgs = flatten(
-                map(lambda p: _do_next_graph_level(p, do_fn), next_pkgs)
-            )
+    Returns a list of Package objects where at least one of the path in the output_paths
+    field matches the search_string.
+    """
+    filtered_package = filter(
+        lambda package: search_path in [p.path for p in package.output_paths],
+        nix_graph.packages,
+    )
+    return list(filtered_package)
```

Comment:
- List comprehensions are more Pythonic
- any with a generator won't compute the entire list before checking

```suggestion
    return [
        package
        for package in nix_graph.packages
        if any(search_path == output_path.path for output_path in package.output_paths)
    ]
```

Reply:
Thanks, that makes sense. :+1: 
I imported `builtins` here and use `builtins.any` to avoid it using `numpy.any` directly.